### DISCUSSION TITLE
Exposing Runway collections to fieldtypes

### DISCUSSION THREAD

**Discussion by benfurfie** [statamic-rad-pack/runway#369] [Help]
I have a database table and model that I am calling into a test site using Runway. The goal is to test whether or not it's possible to allow people to link to an entry in that model.

So for example, we have a list of schools. This is pulled through a model and returned to the front end with a controller separate from Runway. This all works fine.

We need the users to be able to link to a school from Statamic-powered collection entry using something like the Link fieldtype. However, when I have configured it to be visible from the CP, and a handful of fields through a blueprint, it isn't selectable in the Link fieldtype dropdown.

Is there a way in Runway to expose this model to that fieldtype?

**Comment by duncanmcclean** [comment]
Unfortunately, the [Link Fieldtype is hard-coded](https://github.com/statamic/cms/blob/4.x/resources/js/components/fieldtypes/LinkFieldtype.vue#L27) to only allow linking to URLs, entries & assets. There's no way for Runway to hook in and add models there.

**Reply by benfurfie** [reply]
So effectively the only way to do it would be to build my own fieldtype that can pull from model? And skip Runway etc all together?

**Reply by duncanmcclean** [reply]
If all you need is to link a model to an entry, you can add Runway's BelongsTo fieldtype to your entry blueprint.

**Reply by benfurfie** [reply]
It's more something like...

> It may not always be immediately obvious which schools are 'true' grammar schools. Some schools with 'grammar' in their name are not actually the real thing (eg [The Manchester Grammar School](/schools/the-manchester-grammar-school), [Royal Grammar School Guildford](/schools/royal-grammar-school-rgs-guildford)) and similarly, not all grammars feature the word 'grammar' in their name. [Bournemouth School](/schools/bournemouth-school) and [Tiffin School](/schools/tiffin-school-kingston) are just two examples.

So there would be multiple instances of them linking them like entries within a text field, such as a bard field.

**Reply by duncanmcclean** [reply]
Sorry, I think I'm a bit confused. What do you have in the database and where are you trying to add the link (on an entry / on the model)?
